File: src/main/java/com/jme/asset/store/db/entity/asset/FileTypeEntity.java
package com.jme.asset.store.db.entity.asset;

import com.jme.asset.store.db.entity.BaseEntity;
import org.jetbrains.annotations.NotNull;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;

/**
 * The file type entity
 *
 * @author Gomanchuk Mikhail
 */
@Entity
@Table(name = "FILE_TYPE", uniqueConstraints = @UniqueConstraint(columnNames = {
        "name",
        "mime_type",
        "extension"
}, name = "file_type_uniq_index"
)
)
public class FileTypeEntity extends BaseEntity {

    /**
     * Name of type
     */
    @Column(name = "name", nullable = false)
    private String name;

    /**
     * MIME-type
     */
    @Column(name = "mime_type", nullable = false)
    private String mimeType;

    /**
     * Extension
     */
    @Column(name = "extension", nullable = false)
    private String extension;

    /**
     * Get type's name
     *
     * @return type's name
     */
    public @NotNull String getName() {
        return name;
    }

    /**
     * Set type's name
     *
     * @param name type's name
     */
    public void setName(@NotNull final String name) {
        this.name = name;
    }

    /**
     * Get MIME-type of file
     *
     * @return MIME-type of file
     */
    public @NotNull String getMimeType() {
        return mimeType;
    }

    /**
     * Set MIME-type of file
     *
     * @param mimeType MIME-type of file
     */
    public void setMimeType(@NotNull final String mimeType) {
        this.mimeType = mimeType;
    }

    /**
     * Get Extension of file
     *
     * @return extension
     */
    public @NotNull String getExtension() {
        return extension;
    }

    /**
     * Set extension of file
     *
     * @param extension extension
     */
    public void setExtension(@NotNull final String extension) {
        this.extension = extension;
    }
}

File: front-end/src/model/entity/file-type-entity.ts
/**
 * The information about file-types
 *
 * @author Alena Solonevich
 */

export class FileTypeEntity {

  /**
   * The id of the file type
   */
  _id: number;

  /**
   * The name of the file type
   */
  _name: string;

  /**
   * The MIME type of the file type
   */
  _mimeType: string;

  /**
   * The file type extension
   */
  _extension: string;

  constructor(id: number, name: string, mimeType: string, extension: string,) {
    this._id = id;
    this._name = name;
    this._mimeType = mimeType;
    this._extension = extension
  }

  /**
   * Get the id of file type
   */
  get id(): number {
    return this._id;
  }

  /**
   * Get the name of file type
   *
   * @returns {string} the name of file type
   */
  get name(): string {
    return this._name
  }

  /**
   * Get the mime-type of file type
   *
   * @returns {string} the mime-type
   */
  get mimeType(): string {
    return this._mimeType
  }

  /**
   * Get the extension of file type
   *
   * @returns {string} the extension of file type
   */
  get extension(): string {
    return this._extension
  }
}
